sdk/Trace/Sampler/TraceIdRatioBasedSampler.php

```diff
@@ -45,20 +48,27 @@ public function shouldSample(
         ?API\Attributes $attributes = null,
         ?API\Links $links = null
     ): SamplingResult {
+        // TODO: Add config to adjust which spans get sampled (only default from specification is implemented)
         if (null !== $parentContext && ($parentContext->getTraceFlags() & API\SpanContext::TRACE_FLAG_SAMPLED)) {
             return new SamplingResult(SamplingResult::RECORD_AND_SAMPLED, $attributes, $links);
         }
-
-        // TODO: implement as a function of TraceID when specification is ready
-        $decision = (lcg_value() < $this->probability)
-            ? SamplingResult::RECORD_AND_SAMPLED
-            : SamplingResult::NOT_RECORD;
+        /**
+         * Since php can only store up to 
+         */
+        $traceIdLimit = (1 << 60) - 1;
+        $lowerOrderBytes = hexdec(substr($traceId, strlen($traceId) - 15, 15));
+        $traceIdCondition = $lowerOrderBytes < round($this->probability * $traceIdLimit);
+        $decision = SamplingResult::NOT_RECORD;
+        // TODO: Also sample Spans with remote parent
```

Comment:
I was not sure how to detect whether the current span had a remote parent, yet it was asked to include spans with remote parents for sampling in the specs, so I added it as a TODO and will inquire about it once this pull request is made on the original open-telemetry-php repo.